components/tikv_util/src/trace.rs

```diff
@@ -0,0 +1,106 @@
+use kvproto::span as spanpb;
+use minitrace::Collector;
+
+#[repr(u32)]
+pub enum TraceEvent {
```

Comment:
Can we use enums in spanpb directly to avoid code duplication? It's really hard to maintain multiple ones..

Reply:
> Can we use enums in spanpb directly to avoid code duplication? It's really hard to maintain multiple ones..

In current IDEs(Clion for me specifically), spanpb is hard to get error hint in editing time, and, if we got some compiler errors, there is not a single place to check them without these lines.

Reply:
In the current implementation, if event names are modified, we still need to refer to the original protobuf definition to edit and looks like there is no diffierence..

Reply:
It is simply copying definition again for IDE, which is not protobuf code generated intended to be used for I think. We need to maintain two definitions in two places and what's worse if only one of them is changed we may either silently got a wrong tracing result, or a runtime panic, instead of compile time errors.